domains/client/domain-operator/src/tests.rs

```diff
@@ -207,6 +207,774 @@ pub fn generate_evm_account_list(
     }
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn test_evm_domain_create_contracts_with_allow_list() {
```

Comment:
this is a very big test. Can we split it to test multiple cases in different tests ?

Reply:
It involves some code duplication, but yes.